### PR TITLE
Fix #708, expose #732

### LINKED ISSUE #708: A player's health changes along with size

### PR DESCRIPTION
**Pull request by richelbilderbeek**
Me and @OggyOggelito worked on #708 and now the health **is** the diameter; to be precise, the health is the diameter divided by a 100.0.

This fix, however, exposes problems in the collision detection. Fixing that is beyond the fixed Issue, hence we opened a new issue #732, to revise the collision detection and we `IFDEF`-ed away the failing tests.

Good luck reviewing :+1:

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -50,20 +52,17 @@ double player::get_x() const noexcept { return m_c.get_x(); }
...
+  m_health *= m_growth_factor;
 }
 
 /// Make the player shrink
 void player::shrink()
 {
-  m_diameter /= m_growth_factor;
-  if(m_diameter <= m_death_size)
+  m_health /= m_growth_factor;
+  if(this->get_diameter() <= m_death_size)
```
Comment (janclod) [inline src/player.cpp:65]: Remove completely diameter, use only health

Comment (richelbilderbeek) [reply src/player.cpp:65]: `player::get_diameter` converts the health to the diameter, that is why it is called. You suggest a free (as in: non-class non-member) function?

**Review by janclod — APPROVED** [review]
We respect tests so accepted :)

**Comment by TheoPannetier** [comment]
I see PR accepted; I'll take the initiative to merge 👍

**Pull request merged**